### PR TITLE
add unix readline's keymap in INSERT_MODE

### PR DESCRIPTION
**Pull request by curtainp**
when I'm coding, sometime won't exit insert mode frequently for characters move, so this unix `readline` like keymap useful.

### CODE REVIEW

**Comment by JustSage** [comment]
Personally I think this is a really bad habit and I advise against integrating this with insert mode. 
However, this could be useful in command mode.

**Comment by siduck76** [comment]
@curtainp could you add two mappings for moving the cursor <up><down>?

**Comment by siduck76** [comment]
> Personally I think this is a really bad habit and I advise against integrating this with insert mode.
> However, this could be useful in command mode.

might be very helpful for many so I'll add an option to enable /disable this.

File: lua/mappings.lua [hunk 1]
```diff
@@ -24,6 +24,12 @@ map("v", "x", [=[ "_x ]=], opt)
  this line too ]]
 --
 
+-- unix readline's keymap
```
Comment (G-Rowell) [inline lua/mappings.lua:27]: Maybe wrap this in a lua function and check if:
- if the `chadrc.lua` var does not exist, assume this functionality should be disabled (return, early exit the function)
- if the `chadrc.lua` var exists and is true, then run this block

File: lua/chadrc.lua [hunk 2]
```diff
@@ -90,6 +90,12 @@ local M = {
             diffget_3 = "<leader>gl",
             git_blame = "<leader>gb"
         },
+        unix_keymap = {
```
Comment (G-Rowell) [inline lua/chadrc.lua:93]: Function mentioned in my comment below should check the table in `line 9 (above)`, `options`

**Review by G-Rowell — CHANGES_REQUESTED** [review]
Add enable option, assume disable if option is false or does not exist

**Comment by G-Rowell** [comment]
@curtainp 
Lemme know if you want some help or anything!

**Review by G-Rowell — CHANGES_REQUESTED** [review]
This commit should be removed from your branch,

Run `git rebase --interactive HEAD~3`

change `pick` to the delete option for this commit/merge

Then instead of doing the merge to update your branch,
do `git pull --rebase upstream main` (assuming you set siduck76/NvChad as upstream)

This way you avoid adding a merge commit just to update your local branch!

Hope this helps! 

(note, because you are modifying the history of your branch, you'll have to update the online/remote copy with `git push origin --force` once your done)

**Comment by G-Rowell** [comment]
Let me know if you have any questions at all,
happy to help!

**Comment by curtainp** [comment]
@G-Rowell thanks, I have daily somethings bother, the fix I have finished, now to push

**Comment by G-Rowell** [comment]
> @G-Rowell thanks, I have daily things bother, the fix I have finished, now to push 

No problem,
We all have other things we need to do.

@ me when you have pushed your changes or have a question

**Comment by G-Rowell** [comment]
Reading now

**Comment by curtainp** [comment]
> Reading now

there seems somethings wrong, i checkout a new branch to do this, wait me @G-Rowell

**Comment by G-Rowell** [comment]
@curtainp 

Yeah, think you've got some git config issues
Happy to walk you through how setup a PR nicely

**Pull request closed without merging**